sorc/ncep_post.fd/SURFCE.f

```diff
@@ -818,8 +824,12 @@ SUBROUTINE SURFCE
         if(grib=='grib2') then
           cfld=cfld+1
           fld_info(cfld)%ifld=IAVBLFLD(IGET(501))
-          fld_info(cfld)%ntrange=IFHR-ID(18)
-          fld_info(cfld)%tinvstat=1
+            if(ITSRFC>0) then
```

Comment:
@EricJames-NOAA Do you know which version of wgrib2 has dumping errors for  grib2  data generated with these bugs?

Reply:
@WenMeng-NOAA I know for sure the latest version on Hera, which is 3.1.1.  I think it affected some versions earlier than that too.  Let me know if you need me to do some testing to see which versions it affects.

Reply:
@EricJames-NOAA Can you provide me the sample of error message from wgrib2? Somehow my wgrib2 v3.0.2 has no indication of any errors in grib2 data files before and after bug fix. These  fields  with bugs are  for AQ fields. The error message might be used for our user support. Thanks!

Reply:
@WenMeng-NOAA it only happens for the 0-h forecast.  Here's an example of the error message I see with wgrib2 v3.1.1:

[Eric.James@fe1 postprd]$ pwd
/mnt/lfs4/BMC/wrfruc/ejames/RRFS/stmp/tmpnwprd/RRFS_conus_3km/2022102713/postprd
[Eric.James@fe1 postprd]$ /apps/wgrib2/3.1.1/gnu/9.2.0_ncep/bin/wgrib2 RRFS_CONUS.t13z.bgrd3df000.tm00.grib2

pdt_len: bad stat_proc ranges = 0 set to to 1
** ERROR bad grib message: Statistical Processing bad n=0 **
1331:1088575767:d=2022102713:AKHS:surface::

*** FATAL ERROR (delayed): forecast time for RRFS_CONUS.t13z.bgrd3df000.tm00.grib2

Reply:
@EricJames-NOAA Got it. Thanks!